common/acl/manager.go

```diff
@@ -0,0 +1,54 @@
+package acl
+
+import (
+	"io/ioutil"
+
+	"github.com/libp2p/go-libp2p-core/crypto"
+	"k8s.io/klog/v2"
+
+	meshConstants "github.com/kubeedge/edgemesh/common/constants"
+)
+
+type Manager interface {
+	Start()
+	GetPrivateKey() (crypto.PrivKey, error)
+	Name() string
+}
+
+type Type int
+
+const (
+	TypeWithCA Type = iota
+	TypeWithNoCA
+)
+
+var constructMap = make(map[Type]func(config TunnelACLConfig) Manager)
+
+func New(tunnel TunnelACLConfig, t Type) Manager {
+	construct, ok := constructMap[t]
+	if !ok {
+		klog.Fatalf("new ACL manager failed because type %d", t)
+	}
+	m := construct(tunnel)
+	klog.Infof("Use %s to handle acl", m.Name())
+	return m
+}
+
+func NewACLManager(enableSecurity bool, tunnel *TunnelACLConfig) Manager {
```

Comment:
Can enableSecurity be a field of `TunnelACLConfig`? Then reduce one paramter

Reply:
I do not think so. The TunnelACLConfig is just a part of TunnelAgentConfig and TunnelServerConfig, the enable field is the filed of the TunnelAgentConfig and TunnelServerConfig. I would like to keep the tunnelACLConfig just contain the information for acl connection